From a6b8ad565a01d399e9bfa35216a4c2d2b08bc8db Mon Sep 17 00:00:00 2001
From: elupus <elupus@xbmc.org>
Date: Mon, 2 Apr 2012 23:01:42 +0200
Subject: [PATCH 139/145] [dvdplayer] Recalculate codecname when we update
 based on hw (cherry picked from commit
 8033cb5d7b5a6739eb0864d184e25c09ce18e1cf)

---
 .../dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp    |    9 +--------
 .../dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.h      |   14 ++++++++++++--
 2 files changed, 13 insertions(+), 10 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp
index 256bc8b..7ac6b6b 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp
@@ -299,14 +299,7 @@ bool CDVDVideoCodecFFmpeg::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options
   m_pFrame = m_dllAvCodec.avcodec_alloc_frame();
   if (!m_pFrame) return false;
 
-  if(pCodec->name)
-    m_name = CStdString("ff-") + pCodec->name;
-  else
-    m_name = "ffmpeg";
-
-  if(m_pHardware)
-    m_name += "-" + m_pHardware->Name();
-
+  UpdateName();
   return true;
 }
 
diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.h b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.h
index b9cbd9a..b958509 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.h
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.h
@@ -68,8 +68,7 @@ public:
   {
     SAFE_RELEASE(m_pHardware);
     m_pHardware = hardware;
-    m_name += "-";
-    m_name += m_pHardware->Name();
+    UpdateName();
   }
 
 protected:
@@ -79,6 +78,17 @@ protected:
   void FilterClose();
   int  FilterProcess(AVFrame* frame);
 
+  void UpdateName()
+  {
+    if(m_pCodecContext->codec->name)
+      m_name = CStdString("ff-") + m_pCodecContext->codec->name;
+    else
+      m_name = "ffmpeg";
+
+    if(m_pHardware)
+      m_name += "-" + m_pHardware->Name();
+  }
+
   AVFrame* m_pFrame;
   AVCodecContext* m_pCodecContext;
 
-- 
1.7.10

